packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -491,14 +493,25 @@ function updateClassComponent(
       renderExpirationTime,
     );
   }
-  return finishClassComponent(
+  const nextUnitOfWork = finishClassComponent(
     current,
     workInProgress,
     Component,
     shouldUpdate,
     hasContext,
     renderExpirationTime,
   );
+  if (__DEV__) {
+    let inst = workInProgress.stateNode;
+    if (inst.props !== nextProps) {
+      warning(
+        didWarnAboutReassigningProps,
+        'Reassigning `this.props` is not supported and will lead to bugs.',
```

Comment:
I think we can word this message better. `lead to bugs` sounds a bit ambiguous – what bugs? Maybe it's best to link to a docs page or something instead.

Reply:
I don’t know what bugs. Bugs with incorrect props?